### PR TITLE
Add tofu tool and linter

### PR DESCRIPTION
**Pull request by det**
This is a fork of terraform. Just copied the terraform files and replaced terraform with tofu when needed / fixed the download / changed the version / simplified the backwards compat stuff.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,7 +80,7 @@ trunk check enable {linter}
 | SQL             | [sqlfluff], [sqlfmt], [sql-formatter]                                                                                |
 | SVG             | [svgo]                                                                                                               |
 | Swift           | [stringslint], [swiftlint], [swiftformat]                                                                            |
-| Terraform       | [terraform] (validate and fmt), [checkov], [tflint],[tfsec],[terrascan]                                              |
+| Terraform       | [terraform] (validate and fmt), [checkov], [tflint],[tfsec],[terrascan] [tofu]                                       |
```
Comment (TylerJang27) [inline README.md:83]: add comma and fix the rest of the spacing in that line

File: linters/tofu/tofu.test.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { linterCheckTest, linterFmtTest } from "tests";
+import { TrunkLintDriver } from "tests/driver";
+
+// Due to tofu's validate subcommand being disabled by default, we need to manually enable it in our test's trunk.yaml.
+const preCheck = (driver: TrunkLintDriver) => {
+  const trunkYamlPath = ".trunk/trunk.yaml";
+  const currentContents = driver.readFile(trunkYamlPath);
+  const sqlfluffRegex = /- tofu@(.+)\n/;
```
Comment (TylerJang27) [inline linters/tofu/tofu.test.ts:8]: nit: this should be renamed (in the terraform test too 🙃 )

**Review by TylerJang27 — APPROVED** [review]
Looks good, please fix the readme copy

File: linters/tofu/tofu.test.ts (hunk 2 above)
Comment (det) [reply linters/tofu/tofu.test.ts:8]: We have 14 instances of this, will break into its own PR :)

**Pull request merged**